lib/portlayer/task/common.go

```diff
@@ -31,30 +31,38 @@ func toggleActive(op *trace.Operation, h interface{}, id string, active bool) (i
 	if !ok {
 		return nil, fmt.Errorf("Type assertion failed for %#+v", handle)
 	}
-
-	stasks := handle.ExecConfig.Sessions
-	etasks := handle.ExecConfig.Execs
-
-	taskS, okS := stasks[id]
-	taskE, okE := etasks[id]
-
-	if !okS && !okE {
-		return nil, fmt.Errorf("unknown task ID: %s", id)
+	taskS, okS := handle.ExecConfig.Sessions[id]
+	taskE, okE := handle.ExecConfig.Execs[id]
+
+	op.Debugf("target task ID: %s", id)
+	op.Debugf("session tasks during inspect: %s", handle.ExecConfig.Sessions)
+	// print all of them, otherwise we will have to assemble the id list regardless of
+	// the log level at the moment. If there is a way to check the log level we should
+	// do that. since the other approach will slow down all calls to toggleActive.
+	op.Debugf("exec tasks during inspect: %s", handle.ExecConfig.Execs)
+
+	var task *executor.SessionConfig
+	if okS {
+		task = taskS
 	}
 
-	task := taskS
-	if handle.Runtime != nil && handle.Runtime.PowerState != types.VirtualMachinePowerStatePoweredOff {
-		op.Debugf("Task bind configuration applies to ephemeral set")
+	if okE {
```

Comment:
Does it mean exec task gets precedence over Session task?
Why would not you then construct the logic like this:
```
var task *executor.SessionConfig
if taskE, okE := handle.ExecConfig.Execs[id]; okE {
  task = taskE
} else if taskS, okS := handle.ExecConfig.Sessions[id]; okS
  task = taskS
}
```

Reply:
I actually have been fairly unsure if crownding a branching statement with the var assignments is good practice with golang. I have made  the change and the code definitely looks more concise, I avoided this orginally since I was worried about readability.